app/index.js

```diff
@@ -784,6 +784,11 @@ class GlobalBindings {
     this.requestMove = (user, channel) => {
       if (this.connected()) {
         user.model.setChannel(channel.model)
+        // reflect this change in URL
+        let currentUrl = url.parse(document.location.href, true)
+        currentUrl.query.channelName = channel.name()
+        delete currentUrl.search
```

Comment:
This is probably mostly my ignorance of the respective API but why is this necessary?

Reply:
because `search` and `query` parameters point both to the same data (the string after `?`), but `search` is only the full string, whereas `query` is an object. As stated [in the doc](https://github.com/defunctzombie/node-url#urlformaturlobj), if both are provided only `search` is used.